packages/ember-testing/lib/helpers.js

```diff
@@ -57,6 +57,8 @@ function focus(el) {
 
 function visit(app, url) {
   var router = app.__container__.lookup('router:main');
+  router._setupLocation();
```

Comment:
@rwjblue - Yes, like I said, I figured it wasn't the correct way to fix the problem just a place to discuss the fix. :grinning: 

Aha! I see, so `boot()` gets the application up and running, so to speak, but the important thing for our needs is the `advanceReadiness` call which means we get the router + location set up.